From 0fffce9692be7e18d9aefa3273b8b1f70f11fbbc Mon Sep 17 00:00:00 2001
From: Naushir Patuck <naush@raspberrypi.com>
Date: Fri, 20 May 2022 13:18:58 +0100
Subject: [PATCH] media: imx296: Add the link frequency control

Add the V4L2_CID_LINK_FREQ menu control for the imx296. Report a single value
of 1188 Mhz for the link frequency.

Signed-off-by: Naushir Patuck <naush@raspberrypi.com>
---
 drivers/media/i2c/imx296.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/drivers/media/i2c/imx296.c b/drivers/media/i2c/imx296.c
index 64d46badbee7..c1e71ae0442d 100644
--- a/drivers/media/i2c/imx296.c
+++ b/drivers/media/i2c/imx296.c
@@ -308,6 +308,10 @@ static const char * const imx296_test_pattern_menu[] = {
 	"Checks",
 };
 
+static const s64 imx296_link_freq_menu[] = {
+	1188000000ULL,
+};
+
 static void imx296_adjust_exposure_range(struct imx296 *sensor,
 					 struct v4l2_ctrl *ctrl)
 {
@@ -412,7 +416,7 @@ static int imx296_ctrls_init(struct imx296 *sensor)
 	if (ret < 0)
 		return ret;
 
-	v4l2_ctrl_handler_init(&sensor->ctrls, 11);
+	v4l2_ctrl_handler_init(&sensor->ctrls, 12);
 
 	sensor->exposure = v4l2_ctrl_new_std(&sensor->ctrls, &imx296_ctrl_ops,
 					   V4L2_CID_EXPOSURE, 1, 1048575, 1,
@@ -463,6 +467,10 @@ static int imx296_ctrls_init(struct imx296 *sensor)
 				     ARRAY_SIZE(imx296_test_pattern_menu) - 1,
 				     0, 0, imx296_test_pattern_menu);
 
+	v4l2_ctrl_new_int_menu(&sensor->ctrls, NULL, V4L2_CID_LINK_FREQ,
+			       ARRAY_SIZE(imx296_link_freq_menu) - 1, 0,
+			       imx296_link_freq_menu);
+
 	v4l2_ctrl_new_fwnode_properties(&sensor->ctrls, &imx296_ctrl_ops,
 					&props);
 
-- 
2.32.0

